### PR TITLE
Refactor rendering of test results

### PR DESCRIPTION
**Pull request by pietroalbini**
This PR adds a few refactorings to improve the code rendering the test results:

* The template has been moved to a dedicated `templates/` directory.
* The code to render the template has been refactored out of the directive.

### CODE REVIEW

**Review by Urhengulas — APPROVED** [review]
LGTM.

Added one small refactoring suggestion

File: ferrocene/doc/qualification-report/exts/ferrocene_test_outcomes/render_template.py [hunk 1]
```diff
@@ -19,37 +19,45 @@ class RenderOutcomesTemplate(SphinxDirective):
...
+                "target": self.options["target"],
+                "bare_metal_test_target": self.options.get("bare_metal_test_target"),
+                "remote_testing": "remote_testing" in self.options,
+                # Can be None if test outcomes were not injected.
+                "platform_outcomes": self.env.ferrocene_test_outcomes.platform(
+                    self.options["host"],
+                    # Grab the outcomes for the bare metal test target if specified:
+                    self.options["bare_metal_test_target"]
+                    if "bare_metal_test_target" in self.options
+                    else self.options["target"],
```
[the remark this answers is not in the document]
Comment (pietroalbini) [reply ferrocene/doc/qualification-report/exts/ferrocene_test_outcomes/render_template.py]: Good catch!

**Comment by Urhengulas** [comment]
bors merge

**Review by Urhengulas — APPROVED** [review]
bors merge

**Pull request merged**